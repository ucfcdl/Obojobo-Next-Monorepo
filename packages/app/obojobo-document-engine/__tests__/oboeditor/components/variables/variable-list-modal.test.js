import React from 'react'
import { shallow, mount } from 'enzyme'

import VariableListModal from '../../../../src/scripts/oboeditor/components/variables/variable-list-modal'

describe('VariableListModal', () => {
	test('VariableListModal node', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'mock'
				},
				{
					name: 'var2',
					type: 'mock'
				}
			]
		}
		const component = shallow(<VariableListModal content={content} />)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableListModal handle invalid variables', () => {
		const content = {
			variables: null
		}
		const component = shallow(<VariableListModal content={content} />)

		expect(component.find('.single-variable').length).toEqual(0)
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal calls onClickVarible', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'static-value'
				},
				{
					name: 'var2',
					type: 'random-number'
				},
				{
					name: 'var3',
					type: 'static-list'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		component
			.find('.single-variable')
			.at(1)
			.simulate('click')

		expect(
			component
				.find('.single-variable')
				.at(1)
				.hasClass('variable-is-selected')
		).toEqual(true)

		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableListModal clicks on "+ Create Variable" button', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'static-value'
				},
				{
					name: 'var2',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal calls "onClose" when click "OK"', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'static-value'
				},
				{
					name: 'var2',
					type: 'static-value'
				}
			]
		}
		const onClose = jest.fn()
		const component = mount(<VariableListModal content={content} onClose={onClose} />)

		component
			.find('button')
			.at(3)
			.simulate('click')

		expect(onClose).toHaveBeenCalled()
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal duplicates variable correct', () => {
		const content = {
			variables: [
				{
					name: 'var',
					type: 'static-value'
				},
				{
					name: 'mock',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		// Click duplicate button
		component
			.find('button')
			.at(2)
			.simulate('click')

		expect(component.find('.single-variable').length).toEqual(3)
		// Duplicate element is selected
		expect(
			component
				.find('.single-variable')
				.at(2)
				.hasClass('variable-is-selected')
		).toEqual(true)

		expect(
			component
				.find('.single-variable')
				.at(2)
				.html()
		).toMatchInlineSnapshot(
			`"<div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div>"`
		)

		// Click duplicate button
		component
			.find('button')
			.at(2)
			.simulate('click')

		expect(component.find('.single-variable').length).toEqual(4)
		// Duplicate element is selected
		expect(
			component
				.find('.single-variable')
				.at(3)
				.html()
		).toMatchInlineSnapshot(
			`"<div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var3</h4><small><p>static-value</p></small></div>"`
		)

		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal calls onChange', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'static-value'
				},
				{
					name: 'var2',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		// Simulate input changes
		component
			.find('.input-item')
			.at(0)
			.simulate('change', { target: { name: 'name', value: 'mockNewName' } })

		expect(content.variables[0].name).toEqual('mockNewName')
		expect(component.html()).toMatchSnapshot()
	})

	test('onChange resets properties if type is changed', () => {
		const content = {
			variables: [
				{
					name: 'var1',
					type: 'static-value',
					value: '4'
				},
				{
					name: 'var2',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		// Simulate input changes
		component
			.find('.input-item')
			.at(0)
			.simulate('change', { target: { name: 'type', value: 'random-number' } })

		expect(content.variables[0].type).toEqual('random-number')
		expect(content.variables[0].value).toBeUndefined()
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal call onAddVariable', () => {
		const content = {
			variables: [
				{
					name: 'mockName1',
					type: 'static-value'
				},
				{
					name: 'mockName2',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		// Click create new variable and select a type
		component
			.find('button')
			.at(0)
			.simulate('click')
		component
			.find('.new-variable--type-list--single-item')
			.at(0)
			.simulate('click')

		expect(component.find('.single-variable').length).toEqual(3)

		// Click create new variable and select a type
		component
			.find('button')
			.at(0)
			.simulate('click')
		component
			.find('.new-variable--type-list--single-item')
			.at(0)
			.simulate('click')

		expect(component.find('.single-variable').length).toEqual(4)

		expect(component.html()).toMatchSnapshot()
	})

	test('VariableListModal call onDeleteVariable', () => {
		global.confirm = () => true

		const content = {
			variables: [
				{
					name: 'mockName1',
					type: 'static-value'
				},
				{
					name: 'mockName2',
					type: 'static-value'
				}
			]
		}
		const component = mount(<VariableListModal content={content} />)

		// Click create new variable and select a type
		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(component.find('.single-variable').length).toEqual(1)
		expect(component.html()).toMatchSnapshot()
	})
})
