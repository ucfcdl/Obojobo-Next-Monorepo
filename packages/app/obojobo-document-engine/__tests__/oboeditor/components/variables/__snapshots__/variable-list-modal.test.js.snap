// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableListModal VariableListModal call onAddVariable 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$mockName1</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$mockName2</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var</h4><small><p>static-value</p></small></div><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var2\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><form class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Value: </label><input name=\\"value\\" value=\\"\\"></div></form><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal call onDeleteVariable 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$mockName2</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"mockName2\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><form class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Value: </label><input name=\\"value\\" value=\\"\\"></div></form><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal calls "onClose" when click "OK" 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var1</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var1\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><form class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Value: </label><input name=\\"value\\" value=\\"\\"></div></form><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal calls onChange 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$mockNewName</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"mockNewName\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><form class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Value: </label><input name=\\"value\\" value=\\"\\"></div></form><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal calls onClickVarible 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var1</h4><small><p>static-value</p></small></div><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>random-number</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var3</h4><small><p>static-list</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var2\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><div class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Min Value: </label><input type=\\"number\\" name=\\"valueMin\\" value=\\"\\"></div><div class=\\"variable-values--group\\"><label>Max Value: </label><input type=\\"number\\" name=\\"valueMax\\" value=\\"\\"></div><div class=\\"variable-values--group\\"><label>Decimal places: </label><input type=\\"number\\" name=\\"decimalPlacesMin\\" value=\\"\\"><span>to</span><input type=\\"number\\" name=\\"decimalPlacesMax\\" value=\\"\\"></div></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal clicks on "+ Create Variable" button 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var1</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"variable-holder\\"><p>New Variable...</p></div></div><div class=\\"new-variable\\"><div class=\\"new-variable--title\\"><strong>What type of variable do you want to create?</strong></div><form class=\\"new-variable--type-list\\" value=\\"static-value\\"><div class=\\"new-variable--type-list--single-item is-selected\\"><label><input type=\\"radio\\" id=\\"static-value\\" value=\\"static-value\\" checked=\\"\\"><strong for=\\"static-value\\">Static Number of Text Value</strong><label for=\\"static-value\\">Define a single numeric value</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"random-number\\" value=\\"random-number\\"><strong for=\\"random-number\\">Random Number</strong><label for=\\"random-number\\">Generate a single number between min and max</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"static-list\\" value=\\"static-list\\"><strong for=\\"static-list\\">Static List</strong><label for=\\"static-list\\">Define a list of values</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"random-list\\" value=\\"random-list\\"><strong for=\\"random-list\\">Random List</strong><label for=\\"random-list\\">Generate a list of numbers of any size</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"random-sequence\\" value=\\"random-sequence\\"><strong for=\\"random-sequence\\">Random Sequence</strong><label for=\\"random-sequence\\">Generate a list of ascending/descending numbers</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"pick-one\\" value=\\"pick-one\\"><strong for=\\"pick-one\\">Pick One</strong><label for=\\"pick-one\\">Choose one item at random from a list you create</label></label></div><div class=\\"new-variable--type-list--single-item\\"><label><input type=\\"radio\\" id=\\"pick-list\\" value=\\"pick-list\\"><strong for=\\"pick-list\\">Pick List</strong><label for=\\"pick-list\\">Generate a sub-list by randomly choosing from a list</label></label></div></form></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal duplicates variable correct 3`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$mock</h4><small><p>static-value</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var3</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var3\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><form class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Value: </label><input name=\\"value\\" value=\\"\\"></div></form><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`VariableListModal VariableListModal handle invalid variables 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input type=\\"text\\" class=\\"first-tab\\"/><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align:center\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align:center\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">OK</button></div></div></div><input type=\\"text\\" class=\\"last-tab\\"/></div></div></div>"`;

exports[`VariableListModal VariableListModal node 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input type=\\"text\\" class=\\"first-tab\\"/><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align:center\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align:center\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var1</h4><small><p>mock</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>mock</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var1\\"/><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">OK</button></div></div></div><input type=\\"text\\" class=\\"last-tab\\"/></div></div></div>"`;

exports[`VariableListModal onChange resets properties if type is changed 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Variables</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"variable-list-modal\\"><div class=\\"variable-list\\"><div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$var1</h4><small><p>random-number</p></small></div><div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$var2</h4><small><p>static-value</p></small></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center create-variable-button\\"><button class=\\"button\\" contenteditable=\\"false\\">+ Create Variable</button></div></div><div class=\\"variable-properties\\"><div class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"input-item\\" name=\\"name\\" value=\\"var1\\"><div class=\\"help-tip\\"><p>Alphanumeric plus underscore only</p></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></div><div class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></div><div class=\\"variable-values\\"><div class=\\"variable-values--group\\"><label>Min Value: </label><input type=\\"number\\" name=\\"valueMin\\" value=\\"\\"></div><div class=\\"variable-values--group\\"><label>Max Value: </label><input type=\\"number\\" name=\\"valueMax\\" value=\\"\\"></div><div class=\\"variable-values--group\\"><label>Decimal places: </label><input type=\\"number\\" name=\\"decimalPlacesMin\\" value=\\"\\"><span>to</span><input type=\\"number\\" name=\\"decimalPlacesMax\\" value=\\"\\"></div></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-properties--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
